src/PackageManagement/AzureAutomationTools.PackageManagement.psm1

```diff
@@ -573,6 +573,156 @@ function Add-AatPackageVariable {
     }
 }
 
+function New-AatModulesFile {
+    [CmdletBinding(SupportsShouldProcess = $true, ConfirmImpact = 'Low', PositionalBinding = $false)]
+    param (
+        [Parameter(Mandatory = $true)]
+        [ValidateNotNullOrEmpty()]
+        [string]$Name,
+
+        [Parameter()]
+        [switch]$IncludeSamples,
+
+        [Parameter()]
+        [switch]$Force
+    )
+
+    if ($PSCmdlet.ShouldProcess($Name, "Create modules file")) {
+        $Content = [System.Collections.Generic.List[hashtable]]::new()
+
+        if ($IncludeSamples.IsPresent) {
+            Write-Output "`n`tAzureRm.Profile has been added to the example using the `"Add-AatPackageModule -Name 'AzureRM.Profile' -Version '2.7.0'`" cmdlet." | Out-Default
+            $Content += @(
+                [ordered]@{
+                    Name = 'AzureRm.Profile'
+                    Version = '2.7.0'
+                    Package = 'https://devopsgallerystorage.blob.core.windows.net/packages/azurerm.profile.2.7.0.nupkg'
+                },
+                [ordered]@{
+                    Name = 'CustomPackage'
+                    Package = 'https://[storageaccountname].blob.core.windows.net/[ContainerName]/CustomPackage.zip'
+                }
+            )
+        }
+
+        $FilePath = Join-Path -Path (Get-AatPackageFolderPath -Modules) -ChildPath "$Name.json"
+        New-Item -Path $FilePath -Force:$Force
+        if ($Content.Count -gt 0) {
+            $Content | ConvertTo-Json | Set-Content -Path $FilePath
+        }
+        else {
+            Set-Content -Path $FilePath -Value '[]'
+        }
+    }
+}
+
+function Add-AatPackageModule {
+    [CmdletBinding(DefaultParameterSetName = 'WithoutPackage')]
+    param (
+        [Parameter(Mandatory = $true,
+                    Position = 2)]
+        [Parameter(ParameterSetName = 'WithoutPackage')]
+        [Parameter(ParameterSetName = 'WithPackage')]
+        [string]
+        $Name,
```

Comment:
We should probably validated that is *.json - which should also be added to **New-AatAssetsFile**

Optionally we could append it if its not specified.